internal/core/src/bitset/detail/platform/x86/avx2-impl.h

```diff
@@ -1347,11 +1347,19 @@ struct ArithHelperF32<ArithOpType::Mul, CmpOp> {
 template <CompareOpType CmpOp>
 struct ArithHelperF32<ArithOpType::Div, CmpOp> {
     static inline __m256
-    op(const __m256 left, const __m256 right, const __m256 value) {
+    op_special(const __m256 left, const __m256 right, const __m256 value) {
+        // this is valid for the positive denominator, == and != cases.
         // left == right * value
         constexpr auto pred = ComparePredicate<float, CmpOp>::value;
         return _mm256_cmp_ps(left, _mm256_mul_ps(right, value), pred);
     }
+
+    static inline __m256
+    op(const __m256 left, const __m256 right, const __m256 value) {
+        // left / right == value
+        constexpr auto pred = ComparePredicate<float, CmpOp>::value;
+        return _mm256_cmp_ps(_mm256_div_ps(left, right), value, pred);
```

Comment:
I found that the CPI for _mm256_div_ps is 5, which is 10 times that of _mm256_mul_ps. Can we multiply both right and value by -1 and reverse the CmpOp to avoid performance degradation?